src/porepy/geometry/distances.py

```diff
@@ -565,11 +603,37 @@ def segments_polygon(start, end, poly, tol=1e-5):
 
 
 def segment_overlap_segment_set(
-    start, end, start_set, end_set, return_indices=False, tol=1e-5
-):
+    start: np.ndarray,
+    end: np.ndarray,
+    start_set: np.ndarray,
+    end_set: np.ndarray,
+    return_indices: bool = False,
+    tol: float = 1e-5,
+) -> Union[np.ndarray, tuple[np.ndarray, np.ndarray]]:
     """
+    Detects whether a given segment overlaps the segments in a set.
+    The function currently works only for 2D geometries: ``nd == 2``.
+
+    .. note:: The input tolerance should be very small.
+
+    Parameters:
+        start (shape=(nd,)): Start point of a segment.
+        end (shape=(nd,)): End point of a segment.
+        start_set (shape=(nd, num_segments)): Start points of the set of segments.
+        end_set (shape=(nd, num_segments)): End points of the set of segments.
+        return_indices (optional): Whether the function should also return
+          indices overlappings.
```

Comment:
Maybe "indices of overlappings" here instead, given that I understood this correctly.

Reply:
Done.